Mu2eG4/src/toggleProcesses.cc

```diff
@@ -210,8 +218,68 @@ namespace mu2e{
       G4ProcessManager* pmanager     = particle->GetProcessManager();
       // The process manager takes ownership of the process
       // ph->RegisterProcess(rmp, proton); // RegisterProcess only works for known geant4 processes
+
+      // inline G4int AddRestProcess(G4VProcess* aProcess, G4int ord=ordDefault);
+      // inline G4int AddDiscreteProcess(G4VProcess* aProcess, G4int ord=ordDefault);
+      // inline G4int AddContinuousProcess(G4VProcess* aProcess, G4int ord=ordDefault);
+
+      // void SetProcessOrderingToLast(
+      //                               G4VProcess* aProcess,
+      //                               G4ProcessVectorDoItIndex idDoIt
+      //                               );
+      // Set ordering parameter to the last of all processes
+
+      // enum G4ProcessVectorDoItIndex
+      // {
+      //   idxAll = -1,         // for all DoIt/GPIL
+      //   idxAtRest = 0,       // for AtRestDoIt/GPIL
+      //   idxAlongStep = 1,    // for AlongStepDoIt/GPIL
+      //   idxPostStep =2,      // for AlongSTepDoIt/GPIL
+      //   NDoit =3
+      // };
+
+      //G4int AddProcess( G4VProcess* aProcess,
+      //                  G4int ordAtRestDoIt = ordInActive,
+      //                  G4int ordAlongSteptDoIt = ordInActive,
+      //                  G4int ordPostStepDoIt = ordInActive );
+      // Adds a process to the process List
+      // Return values is the index to the List. Negative return value
+      // indicates that the process has not been added due to some errors
+      // The first argument is a pointer to the process.
+      // Successive arguments are ordering parameters of the process in
+      // process vectors. If value is negative, the process is
+      // not added to the corresponding process vector
+
+      // enum G4ProcessVectorOrdering
+      // {
+      //   ordInActive = -1,    // ordering parameter to indicate InActive DoIt
+      //   ordDefault = 1000,   // default ordering parameter
+      //   ordLast    = 9999    // ordering parameter to indicate the last DoIt
+      // };
+
+      // findoing out if the particle can decay (e.g. radioactively or
+      // otherwise) to avoid forcing a single process;
+      // perhaps checking if there is a AtRestProcess already would be
+      // more general
+
+      G4ProcessVector * pVector  = pmanager->GetProcessList();
+      G4VProcess *adecayProcess = nullptr;
+      for( G4int j=0; j<pmanager->GetProcessListLength(); ++j ) {
+        if( ((*pVector)[j]->GetProcessName()).find("Decay") != std::string::npos ) {
+          adecayProcess = (*pVector)[j];
+          break;
+        }
+      }
+      if (adecayProcess == nullptr || particle->GetPDGLifeTime() <0.0 || particle->GetPDGMass() <= 0.0) {
+        // particle is stable
+        pmanager->AddProcess(scp, ordInActive, ordInActive, ordDefault);
+      } else {
+        pmanager->AddProcess(scp, ordDefault, ordInActive, ordDefault);
```

Comment:
Why do we add the special cut process as "at-rest" for unstable particles?   I would think they should be allowed to decay, we don't really need to kill them because of the number of steps?

Reply:
It is more to be able to kill them then anything else. We can remove that for the number of steps but keep it for some special cases if needed.